DbAdapterTestUtilities/Stubs/StubPrimaryKeyValue.cpp

```diff
@@ -74,33 +74,34 @@ namespace systelab { namespace db { namespace test_utility {
 
 	unsigned int StubPrimaryKeyValue::getFieldValuesCount() const
 	{
-		return (unsigned int) m_fieldValues.size();
+		return static_cast<unsigned int>(m_fieldValues.size());
 	}
 
 	IFieldValue& StubPrimaryKeyValue::getFieldValue(unsigned int index) const
 	{
-		if (index < m_fieldValues.size())
-		{
-			return *(m_fieldValues[index]);
-		}
-		else
+		if (index >= m_fieldValues.size())
 		{
-			throw std::runtime_error( "Invalid primary key field index" );
+			throw std::runtime_error("Invalid primary key field index");
+
 		}
+
+		return *(m_fieldValues.at(index));
 	}
 
 	IFieldValue& StubPrimaryKeyValue::getFieldValue(const std::string& fieldName) const
 	{
-		unsigned int nFields = (unsigned int) m_fieldValues.size();
-		for (unsigned int i = 0; i < nFields; i++)
-		{
-			if (m_fieldValues[i]->getField().getName() == fieldName)
+		const auto fieldValue = std::find_if(m_fieldValues.cbegin(), m_fieldValues.cend(),
+			[&fieldName](const std::unique_ptr<IFieldValue>& field)
 			{
-				return *(m_fieldValues[i]);
-			}
+				return field->getField().getName() == fieldName;
+			});
+
+		if (fieldValue != m_fieldValues.cend())
+		{
+			return *(fieldValue->get());
```

Comment:
You don't need the `->get()` to "get" to be able to "get" the reference when using the `*` (deference) operator.
So you could just `return **fieldValue;`  (First * is from iterator to unique_ptr and second * is from unique_ptr to reference)